### PR TITLE
Mathewlenning - Cleaning JFolder

### PR DESCRIPTION
**Pull request by Kubik-Rubik**
Pull Request for Issue #7963.
#### Summary of Changes

Removed small error in the mentioned PR.
#### Testing Instructions

Please merge by code review. All kudos to @Mathewlenning!

### CODE REVIEW

**Comment by brianteeman** [comment]
rtc
<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/10305">issues.joomla.org/joomla-cms/10305</a>.</sub>

File: libraries/joomla/filesystem/folder.php [hunk 1]
```diff
@@ -82,15 +83,15 @@ public static function copy($src, $dest, $path = '', $force = false, $use_stream
 				switch (filetype($sfid))
 				{
 					case 'dir':
+
 						if ($file != '.' && $file != '..')
 						{
-							$ret = self::copy($sfid, $dfid, null, $force);
-
-							if ($ret !== true)
+							if (!self::copy($sfid, $dfid, null, $force))
```
Comment (rdeutz) [inline libraries/joomla/filesystem/folder.php]: this isn't the same, if the return value changes and the result cast to true then you don't get it !== true checks also the type I would prefer as replacement  self::copy($sfid, $dfid, null, $force) !== true

Comment (roland-d) [inline libraries/joomla/filesystem/folder.php]: I agree with @rdeutz here, @Kubik-Rubik can you update this PR accordingly?

**Comment by zero-24** [comment]
Get off RTC for now.<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/10305">issues.joomla.org/joomla-cms/10305</a>.</sub>

**Comment by brianteeman** [comment]
Is someone able to do a code review of this please @mbabker @roland-d @wilsonge <hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/10305">issues.joomla.org/joomla-cms/10305</a>.</sub>

File: libraries/joomla/filesystem/folder.php [hunk 2]
```diff
@@ -101,55 +102,57 @@ public static function copy($src, $dest, $path = '', $force = false, $use_stream
...
-							if ($ret !== true)
-							{
-								return $ret;
-							}
+			switch (filetype($sfid))
+			{
+				case 'dir':
+					if ($file != '.' && $file != '..')
+					{
+						if (self::copy($sfid, $dfid, null, $force, $use_streams))
```
Comment (roland-d) [inline libraries/joomla/filesystem/folder.php]: Same comment as Robert made above applies here. We need a strict type check as in === true

File: libraries/joomla/filesystem/folder.php [hunk 3]
```diff
@@ -229,14 +232,11 @@ public static function create($path = '', $mode = 0755)
 			// If open_basedir is set we need to get the open_basedir that the path is in
 			if ($obd != null)
 			{
+				$obdSeparator = ":";
 				if (IS_WIN)
```
Comment (roland-d) [inline libraries/joomla/filesystem/folder.php]: Codestyle requires a space

File: libraries/joomla/filesystem/folder.php [hunk 4]
```diff
@@ -428,39 +423,33 @@ public static function move($src, $dest, $path = '', $use_streams = false)
 				return JText::sprintf('JLIB_FILESYSTEM_ERROR_FOLDER_RENAME', $stream->getError());
 			}
 
-			$ret = true;
+			return true;
 		}
-		else
+
+		if ($FTPOptions['enabled'] != 1)
```
Comment (roland-d) [inline libraries/joomla/filesystem/folder.php]: Should be better to do strict cast type checking here as well.

**Comment by roland-d** [comment]
@brianteeman Robert and I did check before and requested @Kubik-Rubik to update the PR. I have checked the rest of the PR as well and posted my comments.

**Comment by laoneo** [comment]
Can you please update the branch and solve the conflicts that we can do another round of testing?

**Pull request closed without merging**